lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs

```diff
@@ -471,6 +494,104 @@ impl FilesystemBackend {
             },
         )))
     }
+
+    fn make_client_drive_query_information_response(
+        rdp_req: efs::ServerDriveQueryInformationRequest,
+        file: Option<&FileCacheObject>,
+    ) -> PduResult<RdpdrPdu> {
+        if file.is_none() {
+            return Ok(RdpdrPdu::ClientDriveQueryInformationResponse(
+                efs::ClientDriveQueryInformationResponse {
+                    device_io_response: efs::DeviceIoResponse::new(
+                        rdp_req.device_io_request.clone(),
+                        NtStatus::UNSUCCESSFUL,
+                    ),
+                    buffer: None,
+                },
+            ));
+        }
+        let file = file.unwrap(); // safe because of the if statement above
```

Comment:
I'd rather avoid `unwrap`

```suggestion
        let file = match file {
            Some(file) => file,
            None => {
                return Ok(RdpdrPdu::ClientDriveQueryInformationResponse(
                    efs::ClientDriveQueryInformationResponse {
                        device_io_response: efs::DeviceIoResponse::new(
                            rdp_req.device_io_request.clone(),
                            NtStatus::UNSUCCESSFUL,
                        ),
                        buffer: None,
                    },
                ))
            }
        };
```

Reply:
5bd154eae4971e3e7668087a2594f8e0eff841f6

Reply:
This gets modified in https://github.com/gravitational/teleport/pull/34139:

https://github.com/gravitational/teleport/pull/34139/files#diff-1b8bd3bdf7e3fb4382d5b034f19725c768263110c7648f3199d74fc77558002fR517-R619

Moving this discussion there.

Reply:
a59a44ede95a22ba2fa1c848c78668c87cf1d71d